File: frontend/src/app/models/patient.ts
export interface PatientDoctorUser {
  id: string;
  fullName: string;
  email: string;
}

export interface PatientDoctor {
  id: number;
  applicationUserId: string;
  applicationUser?: PatientDoctorUser;
  specialization: string;
  email?: string;
}

export interface Patient {
  id: number;
  fullName: string;
  gender: string;
  dateOfBirth: string;
  phoneNumber: string;
  email: string;
  address: string;
  medicalNotes: string;
  doctorId?: number | null;
  doctor?: PatientDoctor | null;
  createdAt: string;
}

export interface CreatePatient {
  fullName: string;
  gender: string;
  dateOfBirth: string;
  phoneNumber: string;
  email: string;
  address: string;
  medicalNotes: string;
  doctorId?: number | null;
}

File: backend/Models/Patient.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Patient
    {
        public int Id { get; set; }

        [Required]
        public string FullName { get; set; } = string.Empty;

        [Required]
        public string Gender { get; set; } = string.Empty;

        public DateTime DateOfBirth { get; set; }

        [Required]
        public string PhoneNumber { get; set; } = string.Empty;

        public string Email { get; set; } = string.Empty;

        public string Address { get; set; } = string.Empty;

        public string MedicalNotes { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public int? DoctorId { get; set; }

        public Doctor? Doctor { get; set; }
    }
}